great_expectations/data_context/store/ge_cloud_store_backend.py

```diff
@@ -333,22 +359,34 @@ def _set(  # type: ignore[override]
             response = requests.post(
                 url, json=data, headers=self.headers, timeout=self.TIMEOUT
             )
-            response_json = response.json()
 
-            object_id = response_json["data"]["id"]
-            object_url = self.get_url_for_key((self.ge_cloud_resource_type, object_id))
-            return GeCloudResourceRef(
-                resource_type=resource_type,
-                ge_cloud_id=object_id,
-                url=object_url,
+            value.id_ = self._retrieve_id_from_response(response)
+
+            key: GeCloudIdentifier = self.build_key(
+                name=value.name,
+                id_=value.id_,
             )
+
+            return self.get(key.to_tuple())
+
         # TODO Show more detailed error messages
         except Exception as e:
             logger.debug(str(e))
             raise StoreBackendError(
                 f"Unable to set object in GE Cloud Store Backend: {e}"
             )
 
+    def _retrieve_id_from_response(self, response: requests.Response) -> str:
+        """Pull out the id from the response object json
+
+        Args:
+            response: response to retrieve id from
+
+        Returns:
+            string of id
+        """
+        return response.json()["data"]["id"]
```

Comment:
I think we should take this opportunity to create a custom object that wraps the response and exposes an `id` attribute or `@property`